src/client/pages/Offer/ProductSelector/ProductSlector.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useRef, useMemo } from 'react'
+import styled from '@emotion/styled'
+import { useTextKeys } from 'utils/textKeys'
+import { MEDIA_QUERIES } from 'utils/mediaQueries'
+import { MainProductCard } from './MainProductCard'
+import { AdditionalProductCard } from './AdditionalProductCard'
+
+type Product = {
+  id: string
+  name: string
+  description: string
+  price: string
+  image: string
+}
+
+type ProductSelectorProps = {
+  className?: string
+  mainProducts: Array<Product>
+  additionalProducts: Array<Product>
+}
+
+const getInitialState = ({
+  mainProducts,
+  additionalProducts,
+}: {
+  mainProducts: Array<Product>
+  additionalProducts: Array<Product>
+}): {
+  mainProducts: Record<string, boolean>
+  additionalProducts: Record<string, boolean>
+} => {
+  const getProductIdByValueMap = (products: Array<Product>) =>
+    products.reduce(
+      (result, product) => ({
+        ...result,
+        [product.id]: false,
+      }),
+      {},
+    )
+
+  return {
+    mainProducts: getProductIdByValueMap(mainProducts),
+    additionalProducts: getProductIdByValueMap(additionalProducts),
+  }
+}
+
+export const ProductSelector = ({
+  className,
+  mainProducts,
+  additionalProducts,
+}: ProductSelectorProps) => {
+  const textKeys = useTextKeys()
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const [selectedProductsByCategory, setSelectedProductsByCategory] = useState(
+    getInitialState({ mainProducts, additionalProducts }),
+  )
+
+  const selectedMainProducts = useMemo(() => {
+    return Object.keys(selectedProductsByCategory.mainProducts).filter(
+      (productId) => selectedProductsByCategory.mainProducts[productId],
+    )
+  }, [selectedProductsByCategory])
+
+  return (
+    <Wrapper className={className}>
+      <section>
+        <h2>Main Coverage</h2>
+        <MainCoverageCardGrid>
+          {mainProducts.map(({ id, name, price, description, image }) => {
+            const isTheOnlySelectedMainProduct =
+              selectedMainProducts.length === 1 &&
+              selectedMainProducts[0] === id
+
+            return (
+              <MainProductCard
+                key={id}
+                checkboxRef={
+                  isTheOnlySelectedMainProduct ? inputRef : undefined
+                }
+                title={name}
+                price={price}
+                description={description}
+                image={image}
+                checked={selectedProductsByCategory.mainProducts[id]}
+                onClick={() => {
+                  if (isTheOnlySelectedMainProduct) {
+                    inputRef.current?.setCustomValidity(
+                      textKeys.OFFER_PAGE_MISSING_MAIN_COVERAGE_ERROR(),
```

Comment:
For some weird reason textkeys is not working here. If I hard code the error message I can see it normally when trying to unselect a main product when I have only that one checked. Any ideas?